crates/solver/src/solver.rs

```diff
@@ -83,6 +87,35 @@ pub trait Solver: Send + Sync + 'static {
     fn name(&self) -> &str;
 }
 
+/// The result a given solver achieved in the auction
+pub enum AuctionResult {
+    /// Solution was valid and was ranked at the given place
+    /// Rank 1 means the solver won the competition
+    Ranked(usize),
```

Comment:
nit: Maybe just is_winner? I don't think it is useful to know the position if you are not the winner.